src/components/charts/PercentageStackedBarChart.jsx

```diff
@@ -0,0 +1,114 @@
+import {
+  BarChart, Bar, Tooltip, XAxis, YAxis,
+  CartesianGrid, LabelList,
+} from 'recharts';
+import PropTypes from 'prop-types';
+import React from 'react';
+import _ from 'underscore';
+import ChartsHelper from './ChartsHelper';
+import './PercentageStackedBarChart.less';
+
+const getPercentageDataLabels = chartData => chartData.map(entry => entry.name);
+
+const xAxisStyle = {
+  fontSize: '10px',
+  fontWeight: 600,
+  lineHeight: '1em',
+  letterSpacing: '.02rem',
+  color: '#3283c8',
+};
+
+const labelListStyle = {
+  fill: '#FFFFFF',
+  fontSize: '14px',
+  fontWeight: 600,
+};
+
+class PercentageStackedBarChart extends React.Component {
+  render() {
+    const percentageData = ChartsHelper.getPercentageData(
+      this.props.data,
+      this.props.percentageFixedPoint,
+    );
+    const percentageDataLabels = getPercentageDataLabels(this.props.data);
+    return (
+      <div className="percentage-bar-chart">
+        <BarChart
+          width={580}
+          height={155}
+          data={percentageData}
+          layout="vertical"
+          margin={{
+            top: 28, right: 12, bottom: 8, left: 12,
+          }}
+          barSize={30}
+        >
+          <Tooltip />
+          <CartesianGrid />
+          <XAxis
+            axisLine={false}
+            tickLine={false}
+            ticks={_.range(0, 101, 10)}
+            type="number"
+            domain={[0, 100]}
+            tickMargin={10}
+            style={xAxisStyle}
+            tickFormatter={ChartsHelper.addPercentage}
+          />
+          <YAxis axisLine={false} tickLine={false} dataKey="name" type="category" hide />
+          {
+            percentageDataLabels.map((name, index) => (
+              <Bar
+                key={name}
+                dataKey={name}
+                stackId="a"
```

Comment:
what does this mean?

Reply:
This is to tell recharts library to stack all little bars together... (I am not very into recharts for this component, It might be easier and more readable if we use d3 or manually create SVG... 😂